src/mlpack/core/util/sfinae_utility.hpp

```diff
@@ -16,6 +16,7 @@
 #define MLPACK_CORE_SFINAE_UTILITY
 
 #include <type_traits>
+#include <cstring>
```

Comment:
I think this is included in some other places, is it needed here?

Reply:
This file uses `size_t` but does not define it anywhere. I guess it would be fine (as `size_t` will probably be defined somewhere above it during a compile), I just changed it because I saw a bunch of errors while editing this file.

Reply:
Ah, fair, should we use `<cstdlib>` instead?  `size_t` should be directly defined in that header.